File: backend/app/routers/content.py
from fastapi import APIRouter, HTTPException

from app.schemas.content import (
    GenerateContentRequest,
    GenerateContentResponse,
    ToneCheckRequest,
    ToneCheckResponse,
)

from app.llm.gemini import (
    generate_content,
    check_tone,
)


router = APIRouter(
    prefix="/generate-content",
    tags=["AI Content Generation"],
)


@router.post(
    "/",
    response_model=GenerateContentResponse,
)
async def generate_campaign_content(
    data: GenerateContentRequest,
):

    try:
        draft = generate_content(
            campaign_type=data.campaign_type,
            brief=data.brief,
            language=data.language,
            audience=data.audience,
        )

        return GenerateContentResponse(
            campaign_type=data.campaign_type,
            language=data.language,
            audience=data.audience,
            brief=data.brief,
            draft=draft,
        )

    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Content generation failed: {str(e)}",
        )


@router.post(
    "/tone-check",
    response_model=ToneCheckResponse,
)
async def tone_check(
    data: ToneCheckRequest,
):

    try:
        result = check_tone(
            message=data.message,
            audience=data.audience,
        )

        return ToneCheckResponse(**result)

    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Tone check failed: {str(e)}",
        )

File: frontend/src/api/content.ts
import { apiClient } from "./client";

export interface GenerateContentRequest {
  campaign_type: string;
  brief: string;
  language: string;
  audience: string;
}

export interface GenerateContentResult {
  campaign_type: string;
  language: string;
  audience: string;
  brief: string;
  draft: string;
}

export interface ToneCheckRequest {
  message: string;
  audience: string;
}

export interface ToneCheckResult {
  appropriate: boolean;
  tone: string;
  issues: string[];
  suggestion: string;
}

export async function generateContent(
  data: GenerateContentRequest
): Promise<GenerateContentResult> {
  const response = await apiClient.post<GenerateContentResult>(
    "/generate-content/",
    data
  );

  return response.data;
}

export async function checkTone(
  data: ToneCheckRequest
): Promise<ToneCheckResult> {
  const response = await apiClient.post<ToneCheckResult>(
    "/generate-content/tone-check",
    data
  );

  return response.data;
}